_posts/2023-02-13-semantic-search-solutions.md

```diff
@@ -0,0 +1,106 @@
+---
+layout: post
+title:  "Building a semantic search engine in OpenSearch"
+authors:
+- mshyani
+- nmishra
+- ylwu
+- seanzheng
+- kolchfa
+date: 2023-02-13
+categories:
+ - technical-post
+meta_keywords: semantic search, semantic search solution, deep neural network, natural language queries, semantic search vs keyword search
+meta_description: Learn about the semantic search, its implementation in OpenSearch, and the fundamental concepts you need to conduct deep neural network-based searches.
+
+excerpt: Semantic search helps search engines understand queries. Unlike traditional search, which takes into account only keywords, semantic search also considers their meaning in the search context. Thus, a deep neural network-based semantic search engine has the ability to answer natural language queries in a human-like manner. In this post, you will learn about semantic search and the ways you can implement it in OpenSearch.
+---
+
+Semantic search helps search engines understand queries. Unlike traditional search, which takes into account only keywords, semantic search also considers their meaning in the search context. Thus, a deep neural network-based semantic search engine has the ability to answer natural language queries in a human-like manner. In this post, you will learn about semantic search and the ways you can implement it in OpenSearch. If you’re new to semantic search, continue reading to learn about its fundamental concepts. For information about building pretrained and custom semantic search solutions in OpenSearch, skip to the [Semantic search solutions](#semantic-search-solutions) section.
+
+## Semantic search vs. keyword search
+
+Imagine a dataset of public images with captions. What should the query [**Wild West**](https://en.wikipedia.org/wiki/American_frontier) on such a dataset return? A human would most probably expect it to return images of cowboys, broncos, and rodeos. While these results have no textual overlap with either **wild** or **west**, they are certainly relevant. However, a keyword-based retrieval system such as [BM25](https://en.wikipedia.org/wiki/Okapi_BM25) will not return images of cowboys precisely because there is no text overlap. Clearly, there are aspects of relevance that cannot be measured by keyword similarity. Humans understand relevance in a far broader sense, which involves semantics, contextual awareness, and general world knowledge. Using semantic search, we would like to build systems that can do the same. 
```

Comment:
I think it's fine because the point here is that there is no textual overlap between "wild west" and "cowboy", "bronco" or "rodeo", and we explicitly provide the expected text.